packages/react-instantsearch-core/src/core/createInstantSearchManager.js

```diff
@@ -257,6 +259,68 @@ export default function createInstantSearchManager({
     }
   }
 
+  function hydrateSearchClient(client, results) {
+    if (!results) {
+      return;
+    }
+
+    if (!client._useCache) {
```

Comment:
We need to make sure that the API client squad is aware that we rely on the `_useCache` property. This will likely change in JS client v4.

Wouldn't it be safer to rely on the existence of `addAlgoliaAgent`?

Reply:
Yes it will definitely change inside the V4. We'll ask to create an API to correctly hydrate the cache of the client, rather than those hacks. We could do both.

Reply:
https://github.com/algolia/react-instantsearch/pull/2537/commits/f622ed82e49e4a77e964ee68e793afd83e0dbcd8